interface-impl-v2/src/main/java/org/mule/apikit/implv2/v10/model/QueryStringImpl.java

```diff
@@ -88,6 +92,32 @@ public boolean isFacetArray(String facet) {
     return result;
   }
 
+  @Override
+  public boolean validate(Map<String, List<String>> queryParams) {
+    return validateQueryParams(queryParams, facets(), getRequiredQueryParamsByGroup());
+  }
+
+  private List<QueryStringGroup> getRequiredQueryParamsByGroup() {
+    List<QueryStringGroup> result = new ArrayList<>();
+    for (TypeDeclaration typeDeclaration : getTypeDeclarations()) {
+      List<String> requiredQueryParams = new ArrayList<>();
+      List<String> allQueryParams = new ArrayList<>();
+      Boolean supportAdditionalProperties = true;
+      if (typeDeclaration instanceof ObjectTypeDeclaration) {
+        ObjectTypeDeclaration objectTypeDeclaration = (ObjectTypeDeclaration) typeDeclaration;
+        supportAdditionalProperties = objectTypeDeclaration.additionalProperties();
+        for (TypeDeclaration property : objectTypeDeclaration.properties()) {
+          if (property.required()) {
+            requiredQueryParams.add(property.name());
+          }
+          allQueryParams.add(property.name());
+        }
+      }
+      result.add(new QueryStringGroup(requiredQueryParams, supportAdditionalProperties, allQueryParams));
```

Comment:
a builder for class QueryStringGroup will help here too

Reply:
yep,, I wanted to confirm the design/idea before going further